src/index.tsx

```diff
@@ -148,6 +163,39 @@ class AzureAD extends React.Component<IProps, IState> {
       });
   }
 
+  private checkIfUserAuthenticated() {
+    if (this.state.authenticationState === AuthenticationState.Unauthenticated && this.isLoggedIn()) {
+      const idToken = this.getCacheItem(this.clientApplication.cacheLocation, 'msal.idtoken');
```

Comment:
should this be a class level const?

Reply:
I don't really think so, I think it should be fetched as necessary, not maintained as state. What do you think?

Reply:
Sorry, wasn't clear in the comment. I meant the string `'msal.idtoken'`, it is used in several places as a const so can introduce it as a const

Reply:
oh yeah that's a fair point then, will change